src/taoverse/model/model_tracker.py

```diff
@@ -77,6 +100,29 @@ def on_miner_model_updated(
             model_metadata (ModelMetadata): The latest model metadata of the miner.
         """
         with self.lock:
+            prev_metadata = self.miner_hotkey_to_model_metadata_dict.get(hotkey, None)
             self.miner_hotkey_to_model_metadata_dict[hotkey] = model_metadata
 
+            # If the model was updated, clear the evaluation results since they're no
+            # longer relevant.
+            if prev_metadata != model_metadata:
+                self.miner_hotkey_to_eval_results[hotkey].clear()
+
             bt.logging.trace(f"Updated Miner {hotkey}. ModelMetadata={model_metadata}.")
+
+    def on_model_evaluated(self, hotkey: str, result: EvalResult) -> None:
+        """Notifies the tracker that a model has been evaluated.
+
+        Args:
+            hotkey (str): The miner's hotkey.
+            result (EvalResult): The evaluation result.
+        """
+        with self.lock:
+            eval_results = self.miner_hotkey_to_eval_results[hotkey]
+            eval_results.append(result)
+            if len(eval_results) > self.max_eval_history_len:
+                eval_results.pop(0)
+
+            eval_results.sort(key=lambda x: x.block)
```

Comment:
Although this shouldn't be needed since theoretically blocks only go in one direction, if we believe there is a case where it might be we should do the sort before the pop right? Otherwise you might pop something newer?

Not sure if the test below passes every time due to this or if I am missing something else.

But if you add something for block 2, 3, 4 then add for block 1 next the block 2 would be removed not the 1 right?

Reply:
Good catch.

Updating the test to use 50 inserts caused it to fail.